src/main/java/com/iotiq/application/service/OrderedProductService.java

```diff
@@ -0,0 +1,36 @@
+package com.iotiq.application.service;
+
+import com.iotiq.application.config.ModelMapperUtil;
+import com.iotiq.application.domain.Seller;
+import com.iotiq.application.messages.customer.contact.BasicInfo;
+import com.iotiq.application.messages.orderedproduct.OrderedProductDto;
+import com.iotiq.application.repository.OrderedProductRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class OrderedProductService {
+
+    private final OrderedProductRepository orderedProductRepository;
+
+    public List<OrderedProductDto> getOrderedProducts(Seller seller) {
+        return orderedProductRepository.findAllBySeller(seller).stream().map(
+                orderedProduct -> {
+                    OrderedProductDto orderedProductDto = ModelMapperUtil.map(orderedProduct, OrderedProductDto.class);
+                    orderedProductDto.setOrderDate(orderedProduct.getOrder().getOrderDate());
+                    orderedProductDto.setOrderNumber(orderedProduct.getOrder().getOrderNumber());
+                    orderedProductDto.setDeliveryAddress(orderedProduct.getOrder().getDeliveryAddress());
+                    orderedProductDto.setDeliveryType(orderedProduct.getOrder().getDeliveryType());
+                    orderedProductDto.setDeliveryAddressType(orderedProduct.getOrder().getDeliveryAddressType());
+                    orderedProductDto.setCustomerBasicInfo(ModelMapperUtil.map(orderedProduct.getOrder().getCustomer().getUser().getPersonalInfo(), BasicInfo.class));
```

Comment:
```
                    orderedProductDto.setCustomerBasicInfo(ModelMapperUtil.map(orderedProduct.getOrder().getCustomer().getUser().getPersonalInfo(), BasicInfo.class));
```
Chanining functions like this seems dangerous because if one of them returns null you will get an unhandled NullPointer here. 

Also are these all entities? Then hibernate might be querying the database for each of these methods. We should make sure that's not the case and these are all eagerly fetched. This will prove problematic because this is done in a loop, too.

Reply:
provided null check